tools/pythonpkg/duckdb/experimental/spark/sql/dataframe.py

```diff
@@ -846,9 +846,23 @@ def unionByName(
         |NULL|   4|   5|   6|
         +----+----+----+----+
         """
-        if not allowMissingColumns:
-            raise ContributionsAcceptedError
-        raise NotImplementedError
+        if allowMissingColumns:
+            for col in set(self.columns) - set(other.columns):
+                other = other.withColumn(
+                    col,
+                    Column(ConstantExpression(None)).cast(self.schema[col].dataType),
+                )
+
+            for col in set(other.columns) - set(self.columns):
+                self = self.withColumn(
```

Comment:
This is altering self which I don't think we want to do, this should be a "const" method with no side effects beyond the returned result